utils/nctl/sh/assets/compile_node.sh

```diff
@@ -12,17 +12,26 @@ source "$NCTL"/sh/utils/main.sh
 
 pushd "$NCTL_CASPER_HOME" || exit
 
-while getopts 'd' opt; do 
+unset OPTIND #clean OPTIND envvar, otherwise getopts can break.
+COMPILE_MODE="release" #default compile mode to release.
+
+while getopts 'd' opt; do
     case $opt in
-        d ) export NCTL_COMPILE_TARGET="debug";;
-        * ) echo "nctl-compile only accepts optional flag -d to compile in debug mode."
+        d ) 
+            COMPILE_MODE="debug"
+            ;;
+        * ) 
+            ;; #ignore other cl flags
     esac
 done
 
-if [ "$NCTL_COMPILE_TARGET" = "debug" ]; then
+if [ "$NCTL_COMPILE_TARGET" = "debug" ] || [ "$COMPILE_MODE" = "debug" ]; then
```

Comment:
Can you add a short info to [the docs](https://github.com/casper-network/casper-node/blob/dev/utils/nctl/docs/commands-setup.md?plain=1#L11) on how to switch between compile modes? I think it's worth mentioning that `debug` mode takes preference, i.e., passing `-d` will override the `NCTL_COMPILE_TARGET=release` and vice versa.

Reply:
second adding to docs